src/lib/Invitation/Persistence/Mapper.php

```diff
@@ -0,0 +1,89 @@
+<?php
+
+/**
+ * @copyright Copyright (C) Ibexa AS. All rights reserved.
+ * @license For full copyright and license information view LICENSE file distributed with this source code.
+ */
+declare(strict_types=1);
+
+namespace Ibexa\User\Invitation\Persistence;
+
+use eZ\Publish\Core\Repository\Values\User\Role;
+use Ibexa\Contracts\Core\Repository\Repository;
+use Ibexa\Contracts\Core\Repository\RoleService;
+use Ibexa\Contracts\Core\Repository\UserService;
+use Ibexa\Contracts\Core\Repository\Values\User\Limitation\RoleLimitation;
+use Ibexa\Contracts\Core\Repository\Values\User\Limitation\SectionLimitation;
+use Ibexa\Contracts\Core\Repository\Values\User\Limitation\SubtreeLimitation;
+use Ibexa\Contracts\Core\Repository\Values\User\UserGroup;
+use Ibexa\Contracts\User\Invitation\Invitation;
+use Ibexa\Core\MVC\Symfony\SiteAccess\SiteAccessServiceInterface;
+
+final class Mapper
```

Comment:
You are mixing database results hydration and domain object creation in single class.